src/connect_async.rs

```diff
@@ -0,0 +1,189 @@
+use std::io::{self, Read, Write};
+
+use futures::{Poll, Future, Async};
+use native_tls;
+use native_tls::{HandshakeError, Error, TlsConnector, TlsAcceptor};
+#[allow(deprecated)]
+use tokio_core::io::Io;
+use tokio_io::{AsyncRead, AsyncWrite};
+
+/// A wrapper around an underlying raw stream which implements the TLS or SSL
+/// protocol.
+///
+/// A `TlsStream<S>` represents a handshake that has been completed successfully
+/// and both the server and the client are ready for receiving and sending
+/// data. Bytes read from a `TlsStream` are decrypted from `S` and bytes written
+/// to a `TlsStream` are encrypted when passing through to `S`.
+#[derive(Debug)]
+pub struct TlsStream<S> {
+    inner: native_tls::TlsStream<S>,
+}
+
+/// Future returned from `TlsConnectorExt::connect_async` which will resolve
+/// once the connection handshake has finished.
+pub struct ConnectAsync<S> {
+    inner: MidHandshake<S>,
+}
+
+/// Future returned from `TlsAcceptorExt::accept_async` which will resolve
+/// once the accept handshake has finished.
+pub struct AcceptAsync<S> {
```

Comment:
We don't really have a need for the "accept" side of things, so this could be deleted :D